core/src/main/java/org/elasticsearch/search/aggregations/bucket/range/date/DateRangeAggregationBuilder.java

```diff
@@ -285,7 +281,25 @@ protected DateRangeAggregatorFactory innerBuild(SearchContext context, ValuesSou
             AggregatorFactory<?> parent, Builder subFactoriesBuilder) throws IOException {
         // We need to call processRanges here so they are parsed and we know whether `now` has been used before we make
         // the decision of whether to cache the request
-        Range[] ranges = processRanges(context, config);
+        RangeAggregator.Range[] ranges = processRanges(range -> {
+            DocValueFormat parser = config.format();
+            assert parser != null;
+            double from = range.getFrom();
+            double to = range.getTo();
+            String fromAsString = range.getFromAsString();
+            String toAsString = range.getToAsString();
+            if (fromAsString != null) {
+                from = parser.parseDouble(fromAsString, false, context.getQueryShardContext()::nowInMillis);
+            } else if (Double.isFinite(from)) {
+                from = parser.parseDouble(Long.toString((long) from), false, context.getQueryShardContext()::nowInMillis);
```

Comment:
> we should just have the parser get the string value of the from and to fields in the request and not have the parser try to work out if the input is a string or a number and convert to string at that point 

Talked about this with @colings86 f2f, the problem with parsing all numeric values back to strings in the aggregation parser is that we end up with different Range objects that are no longer `equal` (something we check e.g. in `DateRangeTests`). Also, doing the conversion at parsing will e.g. still leave the door open for current users of the transport client to still use numeric to/from values by accident which we then don't convert. We aggreed on sticking with the current solution.